compiler+runtime/include/cpp/jank/runtime/oref.hpp

```diff
@@ -392,6 +398,12 @@ namespace jank::runtime
     return o;
   }
 
+  template <typename T>
+  constexpr oref<obj::boolean> make_box(bool const b)
```

Comment:
This isn't a specialization! It's just a new template. You'd need `template <>` and `make_box<obj::boolean>` to make this a specialization.

Reply:
Ohh, gotcha. I tried a couple of others things, but since `jank_true`/`jank_false` are both `oref<struct boolean>`, I tried specializing:

```cpp
  template <typename T, typename... Args>
  requires behavior::object_like<T>
  oref<T> make_box(Args &&...args)
```
but since I forward declared the `struct boolean`, I couldn't get cpp to match this template because of `behavior::object_like`.

Not sure if I'm heading in the right direction or completely lost. I am no good at template meta programming lol

Reply:
> Ohh, gotcha. I tried a couple of others things, but since `jank_true`/`jank_false` are both `oref<struct boolean>`, I tried specializing:
> 
> ```c++
>   template <typename T, typename... Args>
>   requires behavior::object_like<T>
>   oref<T> make_box(Args &&...args)
> ```
> 
> but since I forward declared the `struct boolean`, I couldn't get cpp to match this template because of `behavior::object_like`.
> 
> Not sure if I'm heading in the right direction or completely lost. I am no good at template meta programming lol

Yeah, you don't have `boolean` yet. What you do have is `T::obj_type` and `object_type::boolean`, which you could compare.

```cpp
  template <typename T>
  requires (T::obj_type == object_type::boolean)
  oref<T> make_box(bool const b)
```

Reply:
If that doesn't work for you, let me know and I'll jump in and fix it.

Reply:
Ohh, I didn't know you could use `T::obj_type == object_type::boolean` as part of requires, good to know. I added this specialization and it seems to work 👍